tensorflow/compiler/tf2tensorrt/ops/trt_engine_op.cc

```diff
@@ -36,25 +36,32 @@ REGISTER_OP("TRTEngineOp")
     .Attr("InT: list({int8,float16,float32,int32})")
     .Attr("OutT: list({int8,float16,float32,int32})")
     .Attr("input_shapes: list(shape) = []")
+    .Attr("output_shapes: list(shape) = []")
     .Attr("max_cached_engines_count: int = 1")
     .Attr("workspace_size_bytes: int")
     .Attr("precision_mode: {'FP32', 'FP16', 'INT8'}")
     .Attr("calibration_data: string = ''")
     .Attr("use_calibration: bool = true")
     .Input("in_tensor: InT")
     .Output("out_tensor: OutT")
-    // TODO(jie): TF requires concrete output shape for concrete input shapes.
-    // This is tricky for batch dimension, since we cannot ensure which input
-    // would carry the correct batch dimension (for the current stage of the
-    // implementation, we do require all input tensor to carry the same batch
-    // size, but this could change in the future). Hence we disable shape
-    // inference function as a workaround.
-    .SetShapeFn(shape_inference::UnknownShape)
+    .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
```

Comment:
I believe this ShapeFn is useless, that is, it is as good as shape_inference::UnknownShape.
This is because we already set up the output_shapes of the node in the native segment.
A ShapeFn is only useful if given a more concrete input shapes, it can produce a more concrete output shapes. This is because shape inference works like this:
initially collect the shapes for all the nodes; feed the input shapes to the ShapeFn of an operation to see if it can improve the output shapes of the operation. Repeat this step to all operations until no improvement is found. 

In the future when we support dynamic shapes(explicit batch mode), we will probably need a ShapeFn that can produce a more concrete output shapes than the one we initially put on attribute "output_shapes".